test/js/node/zlib/zlib.test.js

```diff
@@ -265,3 +265,142 @@ describe("zlib.brotli", () => {
     }
   });
 });
+
+it.each([
+  "BrotliCompress",
+  "BrotliDecompress",
+  "Deflate",
+  "Inflate",
+  "DeflateRaw",
+  "InflateRaw",
+  "Gzip",
+  "Gunzip",
+  "Unzip",
+])("%s should work with and without `new` keyword", constructor_name => {
+  const C = zlib[constructor_name];
+  expect(C()).toBeInstanceOf(C);
+  expect(new C()).toBeInstanceOf(C);
+});
+
+describe.each(["Deflate", "DeflateRaw", "Gzip"])("%s", constructor_name => {
+  describe.each(["chunkSize", "level", "windowBits", "memLevel", "strategy", "maxOutputLength"])(
+    "should throw if options.%s is",
+    option_name => {
+      // [], // error: Test "-3.4416124249222144e-103" timed out after 5000ms
+      it.each(["test", Symbol("bun"), 2n, {}, true])("%p", value => {
+        expect(() => new zlib[constructor_name]({ [option_name]: value })).toThrow(TypeError);
+      });
+      it.each([Infinity, -Infinity, -2])("%p", value => {
+        expect(() => new zlib[constructor_name]({ [option_name]: value })).toThrow(RangeError);
+      });
+    },
+  );
+});
+
+for (const [compress, decompressor] of [
+  [zlib.deflateRawSync, zlib.createInflateRaw],
+  [zlib.deflateSync, zlib.createInflate],
+  [zlib.brotliCompressSync, zlib.createBrotliDecompress],
+  // [zlib.gzipSync, zlib.createGunzip],
```

Comment:
was this intentionally commented out?

Reply:
for now, yes, and they werent passing before either